htsget-search/src/htsget/bam_search.rs

```diff
@@ -93,6 +96,34 @@ where
   async fn read_raw_header(reader: &mut AsyncReader<ReaderType>) -> io::Result<String> {
     let header = reader.read_header().await;
     reader.read_reference_sequences().await?;
+
+    if let Ok(header) = header.as_deref() {
+      for value in header.split_whitespace() {
+        if let Some(value) = value.strip_prefix("PL:") {
+          if let Err(err) = Platform::from_str(value) {
+            match err {
```

Comment:
I'd keep this match [#[non-exhaustive]](https://rust-lang.github.io/rfcs/2008-non-exhaustive.html) if possible, otherwise we'll get compile warnings every time the spec (and noodles) gets updated :-S

Reply:
Good point, I'll make that an `if let`.